### PR TITLE
[in_app_purchase] Adds additional explanation on the importance of completing a purchase

### PR DESCRIPTION
**Pull request by mvanbeusekom**
Improves the documentation on completing purchases by explaining why it is necessary. This as a request from Zoey came in regarding feedback that it's not clear to developers why completing a purchase is important. 

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [relevant style guides] and ran [the auto-formatter]. (Unlike the flutter/flutter repo, the flutter/plugins repo does use `dart format`.)
- [x] I signed the [CLA].
- [x] The title of the PR starts with the name of the plugin surrounded by square brackets, e.g. `[shared_preferences]`
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I updated `pubspec.yaml` with an appropriate new version according to the [pub versioning philosophy], or this PR is [exempt from version changes].
- [x] I updated `CHANGELOG.md` to add a description of the change, [following repository CHANGELOG style].
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making, or this PR is [test-exempt].
- [x] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/plugins/blob/master/CONTRIBUTING.md
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[relevant style guides]: https://github.com/flutter/plugins/blob/master/CONTRIBUTING.md#style
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat
[pub versioning philosophy]: https://dart.dev/tools/pub/versioning
[exempt from version changes]: https://github.com/flutter/flutter/wiki/Contributing-to-Plugins-and-Packages#version-and-changelog-updates
[following repository CHANGELOG style]: https://github.com/flutter/flutter/wiki/Contributing-to-Plugins-and-Packages#changelog-style
[the auto-formatter]: https://github.com/flutter/plugins/blob/master/script/tool/README.md#format-code
[test-exempt]: https://github.com/flutter/flutter/wiki/Tree-hygiene#tests

### CODE REVIEW

File: packages/in_app_purchase/in_app_purchase/README.md [hunk 1]
```diff
@@ -190,11 +190,15 @@ if (_isConsumable(productDetails)) {
...
-the user, call `InAppPurchase.completePurchase` to tell the App Store
-and Google Play that the purchase has been finished.
+The `InAppPurchase.purchaseStream` will send purchase updates after initiating
+the purchase flow using `InAppPurchase.buyConsumable` or 
+`InAppPurchase.buyNonConsumable`. After verifying the purchase receipt and the 
+delivering the content to the user it ids important to call 
+`InAppPurchase.completePurchase` to tell the App Store and Google Play that the
+purchase has been completed. Calling `InAppPurchase.completePurchase` will 
+inform the App Store and Google Play that the App verified and processed the 
+purchase and Apple and Google can proceed to finalize the transaction and bill
```
Comment (stuartmorgan) [inline packages/in_app_purchase/in_app_purchase/README.md:200]: s/Apple and Google/the store/

File: packages/in_app_purchase/in_app_purchase/README.md [hunk 2]
```diff
@@ -190,11 +190,15 @@ if (_isConsumable(productDetails)) {
...
+the end users payment account.
```
Comment (stuartmorgan) [inline packages/in_app_purchase/in_app_purchase/README.md:201]: Typo: s/users/user's/

**Review by stuartmorgan — APPROVED** [review]
LGTM with nits

File: packages/in_app_purchase/in_app_purchase/README.md [hunk 3]
```diff
@@ -190,11 +190,15 @@ if (_isConsumable(productDetails)) {
...
 
-The `InAppPurchase.purchaseStream` will send purchase updates after
-you initiate the purchase flow using `InAppPurchase.buyConsumable`
-or `InAppPurchase.buyNonConsumable`. After delivering the content to
-the user, call `InAppPurchase.completePurchase` to tell the App Store
-and Google Play that the purchase has been finished.
+The `InAppPurchase.purchaseStream` will send purchase updates after initiating
+the purchase flow using `InAppPurchase.buyConsumable` or 
+`InAppPurchase.buyNonConsumable`. After verifying the purchase receipt and the 
+delivering the content to the user it ids important to call
```
[the remark this answers is not in the document]
Comment (mvanbeusekom) [reply packages/in_app_purchase/in_app_purchase/README.md]: Done.

File: packages/in_app_purchase/in_app_purchase/README.md [hunk 4]
```diff
@@ -190,11 +190,15 @@ if (_isConsumable(productDetails)) {
...
-The `InAppPurchase.purchaseStream` will send purchase updates after
-you initiate the purchase flow using `InAppPurchase.buyConsumable`
-or `InAppPurchase.buyNonConsumable`. After delivering the content to
-the user, call `InAppPurchase.completePurchase` to tell the App Store
-and Google Play that the purchase has been finished.
+The `InAppPurchase.purchaseStream` will send purchase updates after initiating
+the purchase flow using `InAppPurchase.buyConsumable` or 
+`InAppPurchase.buyNonConsumable`. After verifying the purchase receipt and the 
+delivering the content to the user it ids important to call 
+`InAppPurchase.completePurchase` to tell the App Store and Google Play that the
```
[the remark this answers is not in the document]
Comment (mvanbeusekom) [reply packages/in_app_purchase/in_app_purchase/README.md]: Yes that is a good idea, I have replaced these occurrences throughout the document where context allowed and added a definition at the top of the document as well.

File: packages/in_app_purchase/in_app_purchase/README.md [hunk 5]
```diff
@@ -190,11 +190,15 @@ if (_isConsumable(productDetails)) {
...
+purchase has been completed. Calling `InAppPurchase.completePurchase` will 
+inform the App Store and Google Play that the App verified and processed the
```
[the remark this answers is not in the document]
Comment (mvanbeusekom) [reply packages/in_app_purchase/in_app_purchase/README.md]: Done.

[the remark this answers is not in the document]
Comment (mvanbeusekom) [reply packages/in_app_purchase/in_app_purchase/README.md]: Done.

File: packages/in_app_purchase/in_app_purchase/README.md (hunk 1 above)
Comment (mvanbeusekom) [reply packages/in_app_purchase/in_app_purchase/README.md]: Done.

File: packages/in_app_purchase/in_app_purchase/README.md (hunk 2 above)
Comment (mvanbeusekom) [reply packages/in_app_purchase/in_app_purchase/README.md]: Done.

**Pull request merged**